R/settings.R

```diff
@@ -153,10 +163,8 @@ read_config_file <- function(config_file) {
 validate_config_file <- function(config, config_file, defaults) {
   matched <- names(config) %in% names(defaults)
   if (!all(matched)) {
-    warning(
-      "Found unused settings in config '", config_file, "': ", toString(names(config)[!matched]),
-      call. = FALSE
-    )
+    unused_settings <- names(config)[!matched] # nolint: object_usage_linter.
+    cli_warn("Found unused settings in config {.str config_file}: {toString(unused_settings)}")
```

Comment:
do we need `toString()` here?

Reply:
Good catch. That was unintentional. I have gotten rid of it from this file, and also elsewhere in the context of cli.